### PR TITLE
Vulkan: Remove buffer_deprecated

### PR DESCRIPTION
**Pull request by vlj**
Use the simpler buffer class for attrib buffer now. Fix a bug when detecting memory types too.

### CODE REVIEW

File: rpcs3/Emu/RSX/VK/VKVertexBuffers.cpp [hunk 1]
```diff
@@ -457,12 +459,13 @@ VKGSRender::upload_vertex_data()
 						data_size = converted_buffer.size();
 					}
 
-					auto &buffer = m_attrib_buffers[index];
 
-					buffer.sub_data(0, data_size, data_ptr);
-					buffer.set_format(format);
+					size_t offset_in_attrib_buffer = m_attrib_ring_info.alloc<256>(data_size);
+					void* buf = m_attrib_buffers->map(offset_in_attrib_buffer, data_size);
```
Comment (paulsapps) [inline rpcs3/Emu/RSX/VK/VKVertexBuffers.cpp]: Why not have a copy_to_buffer helper that does map/copy/unmap? :)

File: rpcs3/Emu/RSX/VK/VKProgramPipeline.cpp [hunk 2]
```diff
@@ -534,26 +536,28 @@ namespace vk
 				case input_type_texel_buffer:
 				{
 					auto &buffer_view = texel_buffer_views[texel_buffer_index];
-					buffer_view = null_buffer_view();
 
 					auto &buffer = texel_buffers[texel_buffer_index++];
-					buffer.buffer = null_buffer();
+					buffer.buffer = VK_NULL_HANDLE;
```
Comment (kd-11) [inline rpcs3/Emu/RSX/VK/VKProgramPipeline.cpp]: You cant pass VK_NULL_HANDLE to AMD drivers, hence the null_buffer descriptor. You'd have to throw an exception to keep the driver from crashing.

File: rpcs3/Emu/RSX/VK/VKVertexBuffers.cpp (hunk 1 above)
Comment (kd-11) [inline rpcs3/Emu/RSX/VK/VKVertexBuffers.cpp]: I agre with @paulsapps on this one. I think having a buffer update call reduces the verbosity and should help with readablility. Vulkan can be very verbose and hard to follow.

**Comment by vlj** [comment]
will resubmit later
